hplsql/src/main/java/org/apache/hive/hplsql/Exec.java

```diff
@@ -662,6 +660,31 @@ public void registerUdf() {
     udfRegistered = true;
   }
 
+  private static void addJar(ArrayList<String> sql, String dir, String jarName) {
+    if (isNotBlank(jarName)) {
+      sql.add("ADD JAR " + Paths.get(dir, jarName));
+    }
+  }
```

Comment:
You may call `findJarLike` from this method. For me it's a bit odd to pass null value from outside and check it here.

Reply:
done